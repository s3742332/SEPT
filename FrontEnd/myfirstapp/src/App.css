img {
    width: 100%;
}

body {
    height: 100%;
}

* {
    outline: none !important;
}

a {
    text-decoration: none !important;
    color: inherit !important;
}


/* 
* {
    border: 1px solid black;
} */

.navbar {
    background-color: #e3f2fd;
}

.fa.fa-edit {
    color: #18a2b9;
}

.list-group-item.delete:hover {
    cursor: -webkit-grab;
    background-color: pink;
}

.list-group-item.update:hover {
    cursor: -webkit-grab;
    background-color: gainsboro;
}

.list-group-item.board:hover {
    cursor: -webkit-grab;
    background-color: gainsboro;
}

.fa.fa-minus-circle {
    color: red;
}

.landing {
    position: relative;
    /* background: url("../img/showcase.jpg") no-repeat; */
    background-size: cover;
    background-position: center;
    height: 100vh;
    margin-top: -24px;
    margin-bottom: -50px;
}

.landing-inner {
    padding-top: 80px;
}

.dark-overlay {
    background-color: rgba(0, 0, 0, 0.7);
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
}

.card-form {
    opacity: 0.9;
}

.latest-profiles-img {
    width: 40px;
    height: 40px;
}

.form-control::placeholder {
    color: #bbb !important;
}

.pendingItem {
    margin: 1rem;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
    border-radius: 1rem;
    padding: 1rem !important;
    overflow-wrap: break-word;
}

.pendingContainer {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
}

.accountRow {
    padding-bottom: 1rem;
}

.scroll .hero {
    transform: scale(0.98);
}

.hero {
    background-attachment: fixed;
    transition: transform 0.5s ease-in-out;
}

.hero::after {
    content: '';
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    background: linear-gradient( rgba(0, 0, 0, 0.5) 0, rgba(0, 0, 0, 0.3) 50%, rgba(0, 0, 0, 0.1) 100%);
}

.hero .container-fluid {
    z-index: 10;
}

.ant-list-item-extra {
    height: 100% !important;
    width: 30% !important;
}