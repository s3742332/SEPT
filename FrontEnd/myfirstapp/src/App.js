import React, { Component } from "react";
import "./App.css";
import Dashboard from "./components/Dashboard";
import Header from "./components/Layout/Header";
import "bootstrap/dist/css/bootstrap.min.css";
import { BrowserRouter as Router, Redirect, Route, Switch } from "react-router-dom";
import AddPerson from "./components/Persons/AddPerson";
import { Provider } from "react-redux";
import store from "./store";

import Landing from "./components/Layout/Landing";
import Register from "./components/UserManagement/Register";
import Login from "./components/UserManagement/Login";
import PendingUser from "./components/PendingUser";

function App() {
  const isAdmin = true;
  return (
    <Provider store={store}>
      <Router>
        <div className="App">
          <Header />
          <Switch>
            {
              //Public Routes
            }
            {
              isAdmin ?
                <Route exact path="/" component={Dashboard} />
                : <Route exact path="/" component={Landing} />
            }
            {isAdmin &&
              <Route exact path="/pendingusers" component={PendingUser} />
            }
            {!isAdmin &&
              <Route exact path="/register" component={Register} />}
            {/* <Route exact path="/register" component={Register} /> */}
            <Route exact path="/login" component={Login} />

            <Redirect to="/" />
          </Switch>


        </div>
      </Router>
    </Provider>
  );
}

export default App
