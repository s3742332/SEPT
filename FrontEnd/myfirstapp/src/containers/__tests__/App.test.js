import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { App } from '../../App'
import { ItalicOutlined } from '@ant-design/icons'
import { Provider } from 'react-redux'
import store from "../../store";
    
describe('App', () => {
  test('renders without crashing given the required props', () => {
    const props = {
      isFetching: false,
      dispatch: jest.fn(),
      selectedSubreddit: 'reactjs',
      posts: []
    }
    const wrapper = shallow(<Provider store={store}><App {...props} /></Provider>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})