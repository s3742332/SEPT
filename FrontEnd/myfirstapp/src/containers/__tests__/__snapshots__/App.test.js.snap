// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing given the required props 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <App
    dispatch={[MockFunction]}
    isFetching={false}
    posts={Array []}
    selectedSubreddit="reactjs"
  />
</ContextProvider>
`;
